### ISSUE TITLE
Update Makefile targets related to deployment, and evaluate deployment scripts

### ISSUE THREAD

**Issue by balacij** [JacquesCarette/Drasil#2642]
We should probably
* Add basic documentation (if not existing already) about usage of the scripts we have, at the top of each script
* Update the Makefile targets related to deployment 
* Re-evaluate deployment scripts now that we have our own website generator

**Comment by balacij** [comment]
It looks like we need to change the default shell in the Makefile from `/bin/bash` to a more portable variant that looks up `bash` in the PATH, e.g., `/usr/bin/env bash`.

**Comment by balacij** [comment]
@Ant13731 Does the ordering of the `make analysis` and `make graphs` CI steps matter? I'm just curious because right now they're not in the same order in the Test.yaml and Build.yaml scripts.

**Comment by Ant13731** [comment]
No, I don't think `analysis` or `graphs` rely on each other at all, just as long as they are there before deploying. But it may be good to keep the same order.

**Comment by balacij** [comment]
Thanks!

**Comment by Ant13731** [comment]
Also the `rev` command doesn't work in git bash on Windows, so the `make deploy` target doesn't work. I've been using WSL to get around it, but I figured I should mention it.

**Comment by balacij** [comment]
`rev` is just a command that reverses its inputs. I haven't been able to find a decent replacement for it yet for Windows.

As odd as it is, we will need to fix it for the test suite #2698.

**Comment by Ant13731** [comment]
Hmm, I'm not too sure. I don't really know what the stubs are used for, but I can try and switch things around to see if something works.

**Comment by Ant13731** [comment]
I think it might have to stay in some form, as something with the generated code fails. I got this error when I removed `src_stub` completely:
```
cd "build/GlassBR" && stack exec -- "glassbr" 
make[1]: Entering directory 'C:/Users//Documents/GitHub/Drasil/code/build/GlassBR/src/cpp'
doxygen doxConfig
process_begin: CreateProcess(NULL, doxygen doxConfig, ...) failed.
make (e=2): The system cannot find the file specified.
make[1]: *** [Makefile:22: doc] Error 2
```

**Comment by balacij** [comment]
When #2768 is merged, we should add a note somewhere in the README + Wiki about `make help`.

**Comment by JacquesCarette** [comment]
#2768 is merged now - so time to add a note?

**Comment by Ant13731** [comment]
Yep, working on it in #2780

**Comment by balacij** [comment]
It looks like `make help` doesn't show `clean_artifacts`/`cleanArtifacts` targets under 'Cleaning' section. Do you know if we can fix this easily @Ant13731?

**Comment by Ant13731** [comment]
I think the easiest fix might be to separately define those two targets. There's probably something we could do with `HELP_FUN` but I'm not too familiar with the script itself.

**Comment by balacij** [comment]
I gave Cygwin a go on a fresh Windows VM, and it looks like using the Cygwin terminal instead of Git Bash provides a lot more GNU tools out of the box. In particular, `rev` is included in Cygwin.
